src/scanner.cc

```diff
@@ -486,16 +486,15 @@ static void consume_until_v2(string target, State &state) {
   }
 }
 
-static function<u32string(State &)> read_string(Peek pred) {
-  return [=](State & state) {
-    u32string s;
-    consume_while([&](uint32_t c) {
-        auto res = pred(c);
-        if (res) s += static_cast<uint32_t>(c);
-        return res;
-    })(state);
-    return s;
-  };
+static u32string read_string(bool (*cond)(uint32_t), State &state) {
+  u32string s;
+  uint32_t c = PEEK;
+  while (cond(c)) {
+    s += static_cast<uint32_t>(c);
```

Comment:
Isn't `c` already a `uint32_t`? (See line above the while).

I think this is also fine:
```
s += c;
```

Reply:
Yep, I fixed this one up on the move to `c`.